ext/nnstreamer/tensor_converter/tensor_converter_flexbuf.cc

```diff
@@ -0,0 +1,158 @@
+* @see         https://github.com/nnstreamer/nnstreamer
+* @author      Gichan Jang <gichan2.jang@samsung.com>
+* @bug         No known bugs except for NYI items
+*
+*/
+
+#include <glib.h>
+#include <nnstreamer_log.h>
+#include <nnstreamer_plugin_api.h>
+#include "nnstreamer_plugin_api_converter.h"
+#include <flatbuffers/flexbuffers.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+void init_flxc (void) __attribute__ ((constructor));
+void fini_flxc (void) __attribute__ ((destructor));
+#ifdef __cplusplus
+}
+#endif /* __cplusplus */
+
+/** @brief tensor converter plugin's NNStreamerExternalConverter callback */
+static GstCaps *
+flxc_query_caps (const GstTensorsConfig *config)
+{
+  return gst_caps_from_string (GST_FLEXBUF_CAP_DEFAULT);
+}
+
+/** @brief tensor converter plugin's NNStreamerExternalConverter callback */
+static gboolean
+flxc_get_out_config (const GstCaps *in_cap, GstTensorsConfig *config)
+{
+  GstStructure *structure;
+  g_return_val_if_fail (config != NULL, FALSE);
+  gst_tensors_config_init (config);
+  g_return_val_if_fail (in_cap != NULL, FALSE);
+
+  structure = gst_caps_get_structure (in_cap, 0);
+  g_return_val_if_fail (structure != NULL, FALSE);
+
+  /* All tensor info should be updated later in chain function. */
+  config->info.info[0].type = _NNS_UINT8;
+  config->info.num_tensors = 1;
+  if (gst_tensor_parse_dimension ("1:1:1:1", config->info.info[0].dimension) == 0) {
+    ml_loge ("Failed to set initial dimension for subplugin");
+    return FALSE;
+  }
+
+  if (gst_structure_has_field (structure, "framerate")) {
+    gst_structure_get_fraction (structure, "framerate", &config->rate_n, &config->rate_d);
+  } else {
+    /* cannot get the framerate */
+    config->rate_n = 0;
+    config->rate_d = 1;
+  }
+  return TRUE;
+}
+
+/** @brief tensor converter plugin's NNStreamerExternalConverter callback
+ */
+static GstBuffer *
+flxc_convert (GstBuffer *in_buf, gsize *frame_size, guint *frames_in, GstTensorsConfig *config)
+{
+  GstBuffer *out_buf = NULL;
+  GstMemory *in_mem, *out_mem;
+  GstMapInfo in_info;
+  guint mem_size;
+  gpointer mem_data;
+
+  in_mem = gst_buffer_peek_memory (in_buf, 0);
+  if (gst_memory_map (in_mem, &in_info, GST_MAP_READ) == FALSE) {
+    ml_loge ("Cannot map input memory / tensor_converter::flexbuf.\n");
+    return NULL;
+  }
+
+  flexbuffers::Map tensors = flexbuffers::GetRoot (in_info.data, in_info.size).AsMap ();
+  config->info.num_tensors = tensors["num_tensors"].AsUInt32 ();
+
+  if (config->info.num_tensors > NNS_TENSOR_SIZE_LIMIT) {
+    nns_loge ("The number of tensors is limited to %d", NNS_TENSOR_SIZE_LIMIT);
+    goto done;
+  }
+  config->rate_n = tensors["rate_n"].AsInt32 ();
+  config->rate_d = tensors["rate_d"].AsInt32 ();
+  out_buf = gst_buffer_new ();
+  *frame_size = 0;
+  *frames_in = 1;
+
+  for (guint i = 0; i < config->info.num_tensors; i++) {
+    gchar * tensor_key = g_strdup_printf ("tensor_%d", i);
```

Comment:
It looks like a string key for temporal use only. No need to free it?

Reply:
It should be freed.
Fixed in #3132 
Thanks.